pkg/app/app_fetch.go

```diff
@@ -71,7 +71,7 @@ func (a *App) fetch(dstPath string) (string, exec.CmdRunResult) {
 	}
 
 	// if only one fetch, update dstPath for backwards compatibility
-	if len(a.app.Spec.Fetch) == 1 {
+	if len(a.app.Spec.Fetch) == 1 && a.app.Spec.Fetch[0].Path == "" {
```

Comment:
I verified that there must always be a len of 1 for the fetch step, therefore this `[0].Path` check should never fail